src/python/pants/engine/unions.py

```diff
@@ -8,11 +8,10 @@
 from typing import DefaultDict, Iterable, Mapping, TypeVar
 
 from pants.util.frozendict import FrozenDict
-from pants.util.meta import decorated_type_checkable, frozen_after_init
+from pants.util.meta import frozen_after_init
 from pants.util.ordered_set import FrozenOrderedSet, OrderedSet
 
 
-@decorated_type_checkable
```

Comment:
I think you can now delete this

Reply:
It's used within `meta.py` for `frozen_after_init`, although it could probably be inlined there at this point?